build.sbt

```diff
@@ -54,9 +56,14 @@ lazy val calico = project
       "com.armanbilge" %%% "fs2-dom" % Fs2DomVersion,
       "org.typelevel" %%% "shapeless3-deriving" % "3.3.0",
       "dev.optics" %%% "monocle-core" % MonocleVersion,
-      "com.raquo" %%% "domtypes" % "0.16.0-RC3",
       "org.scala-js" %%% "scalajs-dom" % "2.3.0"
-    )
+    ),
+    Compile / sourceGenerators += Def.task {
+      val calicoSourceManaged = (Compile / sourceManaged).value / "domdefs"
+      DomDefsGenerator.cachedGenerate(calicoSourceManaged)
+      val finder: PathFinder = calicoSourceManaged ** "*.scala"
+      finder.get
```

Comment:
Instead of this `PathFinder` trick, can we return the actual list of files generated from the `cachedGenerate` task?

Tangentially, I don't like this caching strategy using `lastScalaDomTypesVersion.txt`. The build already knows the SDT version, and also if we change our code generating sources in `project/` it should be also invalidate the cache. I need to think about a better way to do this.

Reply:
> Instead of this PathFinder trick, can we return the actual list of files generated from the cachedGenerate task?
Yes we can. The reason I didn't do in the first place is I aimed to stay close to the original SDT API. But yeah, since we needed to overwrite enough generator now, I think this is not a real benefit anymore. On the other side this specific trick is simple enough. But if it bothers you, I can change it.

> I don't like this caching strategy using lastScalaDomTypesVersion.txt
I also don't, for the reasons you mentioned. But at this point where we don't have anything generated checked into source control, it's more or less just a bit inconvenient for calico devs. I think for the full solution we would need to hash sources in project/ together with SDT Version, and save that somewhere. Maybe we can even save it into user tmpdir. Impl would be simple enough.

Reply:
> I think for the full solution we would need to hash sources in project/ together with SDT Version, and save that somewhere. Maybe we can even save it into user tmpdir.

The thing is, if the sources in `project/` change, then you have to `reload` the sbt build anyway. So I don't think we need to save this information anywhere at all. Every time you load the build, you have to generate the sources, but that's it, because the only way to change the SDT version or the source generator config is to reload the build.

Reply:
> Every time you load the build, you have to generate the sources once

... when necessary :) Of course, we can just generate them on load unconditionally. But then, when the rest of the build is modified, in the usual modify/reload loop, sources get needlessly generated again and again. Granted, reloads happen much more seldom than compiles, so this has probably the best cost benefit ratio when it comes to solutions requiring no additional state.

Reply:
Yeah, the current scheme is definitely broken. I just did https://github.com/armanbilge/calico/pull/117/commits/1b064b5cc703c266ae60c8ee56b5e9fc32e02de0 and had to manually delete the `lastScalaDomTypesVersion.txt` file which is bad. At the very least it should go into `target/`, so that `clean` makes it go away.

Reply:
I know, it's bad, totally agree. Added a version without cache but with onload generator. Smallest bottleneck is scalafmt of generated sources, maybe we should just get rid of it to speed reloading up. Although, for me it's fast enough.

Reply:
> Smallest bottleneck is scalafmt of generated sources, maybe we should just get rid of it to speed reloading up.

Yes, I am just about to push this ;) completely unnecessary for generated sources. Thanks!